From c8b499620980b5d3cb755b9fedd83d4657fa7c4b Mon Sep 17 00:00:00 2001
From: Randolph Sapp <rs@ti.com>
Date: Mon, 29 Apr 2024 13:44:00 -0500
Subject: [PATCH 039/465] arm64: dts: ti: k3-j721e-main: add clock and pd to
 gpu

Add clock and power domains to the GPU node for J721E. This brings the
dust power domain control back under the driver and allows users to
override the default clock rate for the GPU's core clock.

Signed-off-by: Randolph Sapp <rs@ti.com>
---
 arch/arm64/boot/dts/ti/k3-j721e-main.dtsi | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/ti/k3-j721e-main.dtsi b/arch/arm64/boot/dts/ti/k3-j721e-main.dtsi
index 02d4cdfd54fb..92a9fa5deddf 100644
--- a/arch/arm64/boot/dts/ti/k3-j721e-main.dtsi
+++ b/arch/arm64/boot/dts/ti/k3-j721e-main.dtsi
@@ -2566,8 +2566,13 @@ gpu: gpu@4e20000000 {
 		compatible = "ti,j721e-pvr", "img,pvr-ge8430";
 		reg = <0x4e 0x20000000 0x00 0x80000>;
 		interrupts = <GIC_SPI 24 IRQ_TYPE_LEVEL_HIGH>;
-		power-domains = <&k3_pds 126 TI_SCI_PD_EXCLUSIVE>;
+		power-domains = <&k3_pds 125 TI_SCI_PD_EXCLUSIVE>,
+				<&k3_pds 126 TI_SCI_PD_EXCLUSIVE>;
+		power-domain-names = "firmware", "dust";
 		clocks = <&k3_clks 125 0>;
+		assigned-clocks = <&k3_clks 125 0>;
+		assigned-clock-rates = <750000000>;
+		clock-names = "core";
 	};
 
 	main_mcan0: can@2701000 {
-- 
2.45.1

